core/dbt/include/global_project/macros/materializations/models/materialized_view/materialized_view.sql

```diff
@@ -38,20 +44,28 @@
     -- `BEGIN` happens here:
     {{ run_hooks(pre_hooks, inside_transaction=True) }}
 
-    -- cleanup
-    -- move the existing view out of the way
+    -- determine the scenario we're in: create, full_refresh, alter, refresh data
     {% if existing_relation is none %}
         {% set build_sql = get_create_materialized_view_as_sql(target_relation, sql) %}
-    {% elif full_refresh_mode or existing_relation.type != 'view' %}
+    {% elif full_refresh_mode or not existing_relation.is_view %}
         {% set build_sql = get_replace_materialized_view_as_sql(target_relation, sql, existing_relation, backup_relation, intermediate_relation) %}
+    {% elif config_updates and on_configuration_change == 'apply' %}
+        {% set build_sql = get_alter_materialized_view_as_sql(target_relation, config_updates, sql, existing_relation, backup_relation, intermediate_relation) %}
+    {% elif config_updates and on_configuration_change == 'skip' %}
+        {% set build_sql = '' %}
+        {{ exceptions.warn("Updates were identified and `on_configuration_change` was set to `skip` for `" ~ target_relation ~ "`") }}
+    {% elif config_updates and on_configuration_change == 'fail' %}
+        {{ exceptions.raise_compiler_error("Updates were identified and `on_configuration_change` was set to `fail`") }}
     {% else %}
-        {% set build_sql = get_refresh_data_in_materialized_view_sql(target_relation) %}
+        {% set build_sql = refresh_materialized_view(target_relation) %}
     {% endif %}
 
     -- build model
```

Comment:
@gshank Here's the scenario we discussed.

Note, we use an empty string on `line 55` to denote that we don't want to update the model if there are updates and the user has selected `skip` for `on_configuration_change`.

We condition here on whether `build_sql` is populated so that we don't try to execute an empty string. However, if we don't run any sql statement (which I assume includes prehooks and posthooks), then `line 79` causes an error because we don't have an open transaction.

We skip `line 79` if we have no `build_sql` (again this needs to consider prehooks and posthooks). But when we do that, we get the error `main is not being called during running model`.

Is there a way to avoid the check on main?

Reply:
That check is in _build_run_model_result, which is called at the end of ModelRunner.execute. That references a "result" which is loaded by code in the provider context... see store_result, load_result, _sql_results, etc.  Possibly you could execute a "store_results" call, and provide a name and response?

"store_results" is called at the end of the "statement" macro, and also in core/dbt/include/global_project/macros/materializations/seeds/seed.sql.  The agate table is optional, so perhaps just a name ("main") and a response? ("OK")

Reply:
Thanks for the suggestion. This appears to work. Does this align with what you were thinking? Are there any other updates we should make that would accompany this?

Reply:
Looks good to me. If you're not seeing any errors, I think you're okay.

Reply:
Note that the "store_results" provider method seems to be changing in https://github.com/dbt-labs/dbt-core/pull/7371, so you'll want to be sure to keep on eye on that.